protobuf/lib/src/protobuf/field_set.dart

```diff
@@ -88,8 +88,10 @@ class _FieldSet {
   }
 
   UnknownFieldSet _ensureUnknownFields() {
-    if (_isReadOnly) return _ReadonlyUnknownFieldSet._empty;
-    if (_unknownFields == null) _unknownFields = new UnknownFieldSet();
+    if (_unknownFields == null) {
+      if (_isReadOnly) return UnknownFieldSet().._markReadOnly();
```

Comment:
We could consider having a single static instance of an empty UnknownFieldSet

Reply:
Done.